app/models/admin_blacklisted_email.rb

```diff
@@ -1,12 +1,17 @@
 class AdminBlacklistedEmail < ApplicationRecord
   before_validation :canonicalize_email
+  after_create_commit :remove_invite_requests
 
   validates :email, presence: true, uniqueness: { case_sensitive: false }, email_format: true
 
   def canonicalize_email
     self.email = AdminBlacklistedEmail.canonical_email(self.email) if self.email
   end
 
+  def remove_invite_requests
+    InviteRequest.where(email: self.email).destroy_all
```

Comment:
This should instead find the invite request by the `simplified_email`.

`email` is canonized/simplified here before this callback, so blocking something like `foo+bar@example.com` turns email into `foo@example.com`. InviteRequest on the other hand does that [simplification only on `simplified_email`](https://github.com/otwcode/otwarchive/blob/7e4fdcbcaa1d874b902320c60e9d864198164b70/app/models/invite_request.rb#L9) and leaves `email` the same. So, if you requested an invite for `foo+bar@example.com`, that would be set as the `email` for InviteRequest, so creating a block of `foo+bar@example.com`, canonized to `foo@example.com`, wouldn't find that request to destroy here. (It'd probably be good to have a test for this.)